test/suites/integration/metrics_test.go

```diff
@@ -0,0 +1,87 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration_test
+
+import (
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/samber/lo"
+
+	"github.com/aws/karpenter-provider-aws/test/pkg/environment/common"
+)
+
+var _ = Describe("Metrics", func() {
+	var instanceTypeCount int
+	var availabilityZoneCount int
+	var capacityTypeCount int
+	BeforeEach(func() {
+		azOut, err := env.EC2API.DescribeAvailabilityZones(env.Context, &ec2.DescribeAvailabilityZonesInput{})
+		Expect(err).ToNot(HaveOccurred())
+		availabilityZoneCount = len(azOut.AvailabilityZones)
+
+		itOut, err := env.EC2API.DescribeInstanceTypes(env.Context, &ec2.DescribeInstanceTypesInput{})
+		Expect(err).ToNot(HaveOccurred())
+		instanceTypeCount = len(itOut.InstanceTypes)
+		capacityTypeCount = 3 // on-demand, spot, and reserved
```

Comment:
We shouldn't see anything for pricing or availability for reserved since we don't have any capacity reservations.

Reply:
We should see it, we've enabled the feature so it's reasonable that the availability for these is set to false and pricing is set to 0